core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -1106,22 +1106,38 @@ public void testInterval()
         NodeLocation location = new NodeLocation(1, 1);
         assertThat(expression("INTERVAL '123' YEAR"))
                 .isEqualTo(new IntervalLiteral(location, "123", Sign.POSITIVE, IntervalField.YEAR, Optional.empty()));
+        assertThat(expression("INTERVAL '124' YEARS"))
```

Comment:
I asked @martint about this a while ago, and he said we don't want to do this because the SQL standard doesn't allow it. AFAIK, PostgreSQL works around it by allowing units inside the value, so it's `INTERVAL '5 days'` there, vs `INTERVAL '5' DAY` in Trino.

Reply:
Can you give an example how it changes the UX?

Reply:
https://trino.io/docs/current/functions/datetime.html#interval-functions plural forms

Reply:
That sounds good! But if I had to ask, it means this PR also needs a doc update.

Reply:
To expand on what @nineinchnick said, these represent the *field* that's being specified, not a unit. The field can be one of YEAR, MONTH, DAY, HOUR , MINUTE or SECOND. The generalized syntax for interval is actually `INTERVAL '...' <from-field> TO <end-field>`. Note that interval syntax in Trino is incomplete. For example, it should be possible to say `INTERVAL '30-2' YEAR TO MONTH` to indicate 30 years and 2 months, but that's not currently supported.

I need to look at the first commit in more depth.

Reply:
@martint @nineinchnick I'll drop second commit and leave the first one